packages/next/src/components/FaustProvider.tsx

```diff
@@ -38,12 +38,14 @@ export function FaustProvider<Props = Record<string, unknown>>({
     cacheSnapshot: isNil(authSnapshot) ? undefined : authSnapshot,
   });
 
+  const value = React.useMemo(
```

Comment:
Will introducing this have any side effects?

Reply:
No this is the recommended approach. The provider is dependant on the client which should be immutable object between each renders.